src/constructs/autoscaling/asg.test.ts

```diff
@@ -295,4 +296,85 @@ describe("The GuAutoScalingGroup", () => {
       },
     });
   });
+
+  test("has an instance role created by default with AssumeRole permissions", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      ...defaultProps,
+      app: "TestApp",
+    });
+
+    expect(stack).toHaveResource("AWS::IAM::Role", {
+      AssumeRolePolicyDocument: {
+        Version: "2012-10-17",
+        Statement: [
+          {
+            Action: "sts:AssumeRole",
+            Effect: "Allow",
+            Principal: { Service: { "Fn::Join": ["", ["ec2.", { Ref: "AWS::URLSuffix" }]] } },
+          },
+        ],
+      },
+    });
+  });
+
+  test("passing in an instance role overrides the default", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      ...defaultProps,
+      app: "TestApp",
+      role: new GuInstanceRole(stack, {
+        app: "OverridenAppName",
```

Comment:
Oooh, why change the app in the role?

Reply:
You wouldn't, but thought it was a clear way to illustrate that the role being passed in as a prop would override the default instance role set.